lib/pymedphys/_vendor/apipkg/__init__.py

```diff
@@ -236,24 +236,38 @@ def __getattribute__(self, name):
             # return getattr(getmod(), name)
             try:
                 return getattr(getmod(), name)
-            except (ImportError, ModuleNotFoundError):
+            except (ImportError, ModuleNotFoundError) as exc:
                 # Support inspection rejection
                 if name in ["__file__", "__spec__", "__path__"]:
                     return None
 
                 no_scope_modname = modname.replace("pymedphys._imports.", "")
 
+                from pymedphys._imports import tomlkit
+
+                from pymedphys._dev.paths import DEPENDENCY_EXTRA_PATH
                 from pymedphys._version import __version__
 
+                extra = "user"
+                with open(DEPENDENCY_EXTRA_PATH) as f:
+                    dep_extra_contents = tomlkit.loads(f.read())
+
+                # Suggest extra with minimal num of dependencies
+                for sorted_extra in sorted(
+                    dep_extra_contents, key=lambda k: len(dep_extra_contents[k])
+                ):
+                    if no_scope_modname in dep_extra_contents[sorted_extra]:
+                        extra = sorted_extra
+                        break
+
                 raise ModuleNotFoundError(
                     f"""
                     PyMedPhys was unable to import "{no_scope_modname}.{name}".
-                    The easiest way to fix this issue is to use the "[user]"
-                    option when installing PyMedPhys. For example,
-                    with pip this can be done by calling
-                    "pip install pymedphys[user]=={__version__}".
+                    The easiest way to fix this issue is to use the "[extra]"
```

Comment:
Do you need to parameterise the extra here? When testing this locally what does it print out if you uninstall a package?

Reply:
This is the printed message:

![Exception_pymedphys](https://github.com/pymedphys/pymedphys/assets/101093472/d282b625-17f5-45ce-b275-46ff532d66ac)

I didn't want to parametrize the first "extra" to keep a similar error message.

Reply:
Beautiful ❤️